src/DDoSX/WafClient.php

```diff
@@ -26,14 +33,141 @@ public function create(Waf $waf)
     {
         $response = $this->post(
             'v1/domains/' . $waf->name . '/waf',
-            json_encode($this->friendlyToApi($waf, $this->requestMap))
+            json_encode($this->friendlyToApi($waf, self::WAF_MAP))
         );
         $body = $this->decodeJson($response->getBody()->getContents());
 
         return (new SelfResponse($body, "domain_name"))
             ->setClient($this)
             ->serializeWith(function ($response) {
-                return new Waf($this->apiToFriendly($response->data, $this->requestMap));
+                return new Waf($this->apiToFriendly($response->data, self::WAF_MAP));
             });
     }
+
+    /**
+     * Gets the WAF settings for the domain
+     * @param $domainName
+     * @return Waf
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getByDomainName($domainName)
+    {
+        $response = $this->get('v1/domains/' . $domainName . '/waf');
+
+        $body = $this->decodeJson($response->getBody()->getContents());
+
+        return $this->serializeWaf($body->data);
+    }
+
+    /**
+     * Get a paginated response of rules for a domain
+     * @param $domainName
+     * @param $page
+     * @param $perPage
+     * @param array $filters
+     * @return \UKFast\SDK\Page
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getRules($domainName, $page = 1, $perPage = 15, $filters = [])
```

Comment:
Should these unctions be in their own client, as they're dealing with different resources? Full CRUD on this could make this a huge class